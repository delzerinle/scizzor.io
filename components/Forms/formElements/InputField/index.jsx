import PropTypes from 'prop-types';
import { Field, ErrorMessage } from 'formik';

const InputField = ({ name, label, icon, type = 'text' }) => {
  const wrapperClass = icon
    ? 'flex border w-full border-alt-2 mb-4 form-input'
    : 'border w-full border-alt-2 mb-4 form-input';
  const inputWrapperClass = icon
    ? 'flex-1 -ml-2 px-4 relative'
    : 'px-4 relative';
  return (
    <div className={wrapperClass}>
      {icon && <div className="pl-4 flex items-center">{icon}</div>}
      <div className={inputWrapperClass}>
        <Field
          id={name}
          type={type}
          name={name}
          className="w-full py-2 form-input__field focus:outline-none relative z-10 bg-transparent font-medium"
        />
        <label
          className="text-secondary font-light tracking-normal absolute py-2 px-4 left-0 transition-all transition-ease z-0 form-input__label"
          htmlFor={name}
        >
          {label}
        </label>
      </div>
    </div>
  );
};

InputField.propTypes = {
  type: PropTypes.string,
  icon: PropTypes.node,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default InputField;
