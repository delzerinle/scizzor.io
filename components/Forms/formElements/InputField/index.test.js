import React from 'react';
import { shallow, mount } from 'enzyme';

import InputField from './index';

describe('Input field tests', () => {
  it('renders without crashing', () => {
    shallow(<InputField name="test" label="test" />);
  });

  it('input field snapshot', () => {
    const wrapper = shallow(<InputField name="test" label="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('input field with icon snapshot', () => {
    const wrapper = shallow(
      <InputField name="test" label="test" icon="test" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
