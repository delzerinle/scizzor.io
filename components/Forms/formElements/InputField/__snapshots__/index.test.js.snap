// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field tests input field snapshot 1`] = `
<div
  className="border w-full border-alt-2 mb-4 form-input"
>
  <div
    className="px-4 relative"
  >
    <Field
      className="w-full py-2 form-input__field focus:outline-none relative z-10 bg-transparent font-medium"
      id="test"
      name="test"
      type="text"
    />
    <label
      className="text-secondary font-light tracking-normal absolute py-2 px-4 left-0 transition-all transition-ease z-0 form-input__label"
      htmlFor="test"
    >
      test
    </label>
  </div>
</div>
`;

exports[`Input field tests input field with icon snapshot 1`] = `
<div
  className="flex border w-full border-alt-2 mb-4 form-input"
>
  <div
    className="pl-4 flex items-center"
  >
    test
  </div>
  <div
    className="flex-1 -ml-2 px-4 relative"
  >
    <Field
      className="w-full py-2 form-input__field focus:outline-none relative z-10 bg-transparent font-medium"
      id="test"
      name="test"
      type="text"
    />
    <label
      className="text-secondary font-light tracking-normal absolute py-2 px-4 left-0 transition-all transition-ease z-0 form-input__label"
      htmlFor="test"
    >
      test
    </label>
  </div>
</div>
`;
